### PR TITLE
Modal updates for accessibility

### PR DESCRIPTION
**Pull request by jbalboni**
Related to [205](https://github.com/department-of-veterans-affairs/sunsets-team/issues/205)

This adds some updates to make our modal more accessible. Mostly, this means focus handling updates. The changes largely follow [this article](https://www.smashingmagazine.com/2014/09/making-modal-windows-better-for-everyone/).

- Focus on modal when it opens, so content is presented to users
- Save the last focused element before the modal, so a user doesn't lose their spot
- Constrain focus changes to be inside the modal while it's open
- Add aria attributes

These all seemed reasonable to me, but if they're actually unhelpful changes, please let me know.

### CODE REVIEW

**Comment by ayaleloehr** [comment]
cc @webinista and @U-DON since this is a common component.

**Comment by webinista** [comment]
LGTM.

**Comment by U-DON** [comment]
LGTM.

File: src/js/common/components/Modal.jsx [hunk 1]
```diff
@@ -1,10 +1,38 @@
 import React from 'react';
 import classNames from 'classnames';
 
+function focusListener() {
+  const listener = event => {
+    const modal = document.querySelector('.va-modal');
+    if (!modal.contains(event.target)) {
+      event.stopPropagation();
+      modal.focus();
```
Comment (jbalboni) [inline src/js/common/components/Modal.jsx]: @U-DON If there were multiple visible, yes, but that wouldn't work with how the modal is displaying right now either. The `va-modal` class only appears to be in the DOM when the modal is visible.

Comment (U-DON) [inline src/js/common/components/Modal.jsx]: Actually, never mind. When the other modals are hidden, they're just a plain `div` without the `va-modal` class. Should be good.

**Pull request merged**